src/uwtools/utils/api.py

```diff
@@ -25,10 +32,127 @@ def ensure_data_source(
     return str2path(data_source)
 
 
+def make_execute(driver_class: type[Driver], with_cycle: bool) -> Callable[..., bool]:
+    """
+    Returns a function that executes tasks for the given driver.
+
+    :param driver_class: The driver class whose tasks to execute.
+    :param with_cycle: Does the driver's constructor take the 'cycle' parameter?
+    """
+
+    def execute(  # pylint: disable=unused-argument
+        task: str,
+        config: Optional[Union[Path, str]] = None,
+        batch: bool = False,
+        dry_run: bool = False,
+        graph_file: Optional[Union[Path, str]] = None,
+        stdin_ok: bool = False,
+    ) -> bool:
+        return _execute(
+            driver_class=driver_class,
+            task=task,
+            cycle=None,
+            config=config,
+            batch=batch,
+            dry_run=dry_run,
+            graph_file=graph_file,
+            stdin_ok=stdin_ok,
+        )
+
+    def execute_cycle(  # pylint: disable=unused-argument
+        task: str,
+        cycle: dt.datetime,
+        config: Optional[Union[Path, str]] = None,
+        batch: bool = False,
+        dry_run: bool = False,
+        graph_file: Optional[Union[Path, str]] = None,
+        stdin_ok: bool = False,
+    ) -> bool:
+        return _execute(
+            driver_class=driver_class,
+            task=task,
+            cycle=cycle,
+            config=config,
+            batch=batch,
+            dry_run=dry_run,
+            graph_file=graph_file,
+            stdin_ok=stdin_ok,
+        )
+
+    execute_cycle.__name__ = "execute"
+    assert _execute.__doc__ is not None
```

Comment:
Is this one just to make sure we don't get odd failures in `re.sub()` calls if someone decided to remove the docstring on the function `_execute()` below?

Reply:
It's actually for the typechecker. Without this assertion, the line after this one is flagged:
```
uwtools/utils/api.py:84: error: Argument 3 to "sub" has incompatible type "str | None"; expected "str"  [arg-type]
        execute_cycle.__doc__ = re.sub(r"\n *:param driver_class:.*\n", "\n", _execute.__doc__)
                                                                              ^~~~~~~~~~~~~~~~
```
I'm surprised, because it seems like static analysis could see that `_execute()` _has_ a docstring so that, although officially its type is technically `str` or `None`, it's a `str` in this case. My guess is that `__str__` may be attached to the class only at runtime: From a syntax perspective, the docstring is just a string literal given as the first statement in the class body. The fact that it's interpreted as a "docstring", and then attached to the `__str__` attribute, may not be known until runtime.